### ISSUE TITLE
Cannot read property 'length' of undefined

### ISSUE THREAD

**Issue by baqirfarooq** [yajra/laravel-datatables#919]
### Code snippet of problem
```
  var employeeTable =  $('#employee-table').DataTable({
                processing: true,
                serverSide: true,
                ajax: {
                    url: '{{route('admin.employee.datatable')}}',
                    data: function (d) {
                        d.s = $('input[name=dtGlobalSearch]').val();
                    }
                },
                scrollX: true,
                sScrollX: "100%",
                buttons: [
                    {extend: 'print', className: 'btn dark btn-outline'},
                    {extend: 'copy', className: 'btn red btn-outline'},
                    {extend: 'pdf', className: 'btn green btn-outline'},
                    {extend: 'excel', className: 'btn yellow btn-outline '},
                    {extend: 'csv', className: 'btn purple btn-outline '},
                    {extend: 'colvis', className: 'btn dark btn-outline', text: 'Columns'}
                ],
                scrollCollapse: true,
                fixedColumns: {
                    leftColumns: 1,
                    rightColumns: 1
                },

                columns: [
                    { data: 'id', name: 'id', width: "5%" },
                    { data: 'name', name: 'name', searchable: true, width: "15%" },
                    { data: 'code', name: 'code', width: "10%" },
                    { data: 'email', name: 'email', width: "10%" },
                    { data: 'mac_address', name: 'mac_address', width: "15%" },
                    { data: 'mobile_number', name: 'mobile_number', width: "15%" },
                    { data: 'status', name: 'status', orderable: false, searchable: false, className: 'text-center', width: "10%"},
                    { data: 'action', name: 'action', orderable: false, searchable: false, className: 'text-center', width: "20%"}
                ],
                "dom": "<'row' <'col-md-12'B>><'row'<'col-md-6 col-sm-12'l><'col-md-6 col-sm-6'f>r><'table-scrollable't><'row'<'col-md-7 col-sm-12'p>>", // horizobtal scrollable datatable

            });
```

**Comment by baqirfarooq** [comment]
i have getting problem in pagination start tu paginate work fine then show 
Uncaught TypeError: Cannot read property 'length' of undefined

**Comment by baqirfarooq** [comment]
in start two paginate any of theme work fine then show error

**Comment by baqirfarooq** [comment]
`  $employees = $this->employee->latest(); `
`

        $datatables = app('datatables')->of($employees)->setTotalRecords($this->employee->count())->addColumn('status', function ($employee) use ($request) {
            $actions = \Html::decode(btn_status('admin.employee.status', $employee->id, $employee->status));
            return $actions;

        })->addColumn('action', function ($employee) use ($request) {
            $actions = '<a href="' . route('admin.employee.show', $employee->id) . '" class="btn btn-xs blue"><i class="fa fa-eye"></i></a>  <a href="' . route('admin.employee.edit', $employee->id) . '" class="btn btn-xs blue"><i class="fa fa-edit"></i> </a><a href="' . route('admin.employee.destroy', $employee->id) . '" class="btn btn-xs red btnDelete"><i class="fa fa-trash-o"></i></a>';
            return $actions;

        });

        if ($datatables->request->has('s')) {
            $employees->where('name', 'like', "%" . $datatables->request->get('s') . "%");
        }

        return $datatables->make(true);`

**Comment by yajra** [comment]
> Uncaught TypeError: Cannot read property 'length' of undefined

I think this issue is related to javascript. Try removing some of the dataTable plugins you used to pin down what's causing the issue

**Comment by SturmB** [comment]
Okay, this is now a serious problem. I'm having the exact same problem, but _only_ on my production Ubuntu server. On my local dev machine, everything works as expected.

The data is being retrieved and displayed in the table, but **pagination** isn't working and it's throwing that `Uncaught TypeError: Cannot read property 'length' of undefined` error message. The code is _identical_ on both machines, so why is this error _only_ appearing on the Ubuntu server?

**Comment by SturmB** [comment]
Yes, I did. I disabled `rowGroup`, `buttons`, `colReorder`, and `responsive`. The problem persists.

**Comment by SturmB** [comment]
Got it. After some further digging, I discovered that the JavaScript error was occurring in DebugBar. Searching for that, I stumbled upon [this bug report](https://github.com/barryvdh/laravel-debugbar/issues/544) that describes my problem. It sounded much more likely that I configured something wrong in nginx rather than there being something in my site's code.

So, doing further digging, I saw my fatal flaw: A missing question mark. The `location` section of my site's nginx configuration was:
```
	location / {
		try_files $uri $uri/ /index.php$query_string;
	}
```
So I changed the `index.php` portion to read `/index.php?$query_string;`. That fixed the issue. Then I spoke with an nginx expert who recommended that I _not_ hard-code the question mark in there, and instead use `/index.php$is_args$args;`, which only looks for the question mark if there are arguments to process. So that's what I'm using now and it works. Yay!

**Wow**, what a long debugging process this was. Sorry for assuming your plugin was the culprit.
